### ISSUE TITLE
Integrate video download status tracking with database backed content caches

### ISSUE THREAD

**Issue by rtibbles** [learningequality/ka-lite#4507]
Now that we have a database for this kind of thing - let's use it!

Deprecate old content download tracking, and use new database instead.

@jamalex Might need some input on design here - it might be helpful to annotate every node with 'download size', 'on disk size' and 'status' (including Topic nodes) in order to be able to do efficient loading of the data (i.e. we can just query one level of the topic tree at a time, rather than the whole thing).

**Comment by jamalex** [comment]
Note that one of the current roles of `VideoFile` is tracking the actual
live download percentage _within_ a video (we update the DB within the
download progress callbacks). The other main role is queuing up files for
download to start with. It's not a great way to do either of those things.

One thing we've never found a good design for is knowing what to put in the
"out of X videos" part of the status, if people a) refresh the page or b)
start queuing additional videos to download while the previous batch is
still going (since we don't keep track of which videos were downloaded
within the current "session", just which ones have yet to download).

We might want to move the "download progress" reporting stuff out of the DB
and into a static JSON data file that gets written by the download process
and read by the browser. That way we don't need to write to the DB every
time there's a small percentage change in the progress of the currently
downloading file. As long as there's only ever one downloading process, we
also wouldn't need to worry about write concurrency. JSON file could look
something like:

```javascript
{
    "total_items": 119,
    "current_index": 53,
    "current_id": "72ec62_Rf10",
    "current_title": "Dividing 0 by 0",
    "current_percent": 13
}
```

Just brainstorming future directions now: I'm thinking it might be good to
have this be part of a prelim version of a more general "task queue"
system. So, we could have multiple "tasks", each with their own ID, and a
separate "progress" JSON file would be written for each task. This would
replace chronograph. Something like Celery might have been a good option
for things like this, but I think I saw a note from @benjaoming that it had
tricky dependencies.

Happy to chat more! Short-term, it may be simplest to just merge what is
currently done with VideoFile into the content cache DBs, but it'll be good
to plan towards the longer-term.

**Comment by rtibbles** [comment]
Fixed in #4630
